src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,28 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  closeModal: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, closeModal }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then(setUser)
+      .catch(() => setUser(null));
```

Comment:
So, as I understood, there will be infinite loading if we don't find any user. One possible solution is to create a new state for loading.